src/vehicle_gui.cpp

```diff
@@ -3084,24 +3084,6 @@ struct VehicleViewWindow : Window {
 	void OnPaint() override
 	{
 		const Vehicle *v = Vehicle::Get(this->window_number);
-		bool is_localcompany = v->owner == _local_company;
-		bool refittable_and_stopped_in_depot = IsVehicleRefittable(v);
-
-		this->SetWidgetDisabledState(WID_VV_RENAME, !is_localcompany);
-		this->SetWidgetDisabledState(WID_VV_GOTO_DEPOT, !is_localcompany);
-		this->SetWidgetDisabledState(WID_VV_REFIT, !refittable_and_stopped_in_depot || !is_localcompany);
-		this->SetWidgetDisabledState(WID_VV_CLONE, !is_localcompany);
-
-		if (v->type == VEH_TRAIN) {
-			this->SetWidgetLoweredState(WID_VV_FORCE_PROCEED, Train::From(v)->force_proceed == TFP_SIGNAL);
-			this->SetWidgetDisabledState(WID_VV_FORCE_PROCEED, !is_localcompany);
-		}
-
-		if (v->type == VEH_TRAIN || v->type == VEH_ROAD) {
-			this->SetWidgetDisabledState(WID_VV_TURN_AROUND, !is_localcompany);
-		}
-
-		this->SetWidgetDisabledState(WID_VV_ORDER_LOCATION, v->current_order.GetLocation(v) == INVALID_TILE);
```

Comment:
Instead of moving all this, create a new method called `UpdateButtons`, and move the `void OnPaint() override` line down. Then call `this->UpdateButtons()` where you need.